### PR TITLE
Change AssertionErrors to Runtime and ValeErrors

### PR DESCRIPTION
**Pull request by constantinpape**

### CODE REVIEW

File: src/python/module/z5py/attribute_manager.py [hunk 1]
```diff
@@ -13,21 +13,21 @@ def __init__(self, path, is_zarr):
 
     def __getitem__(self, key):
```
Comment (clbarnes) [inline src/python/module/z5py/attribute_manager.py]: Is this for compatibility with the C++ layer? It's a `KeyError` in h5py, which is more in line with python conventions around item getting.

File: src/python/module/z5py/dataset.py [hunk 2]
```diff
@@ -230,18 +231,19 @@ def __len__(self):
     def index_to_roi(self, index):
 
         # check index types of index and normalize the index
-        assert isinstance(index, (slice, tuple)), \
-            "z5py.Dataset: index must be slice or tuple of slices"
+        if not isinstance(index, (slice, tuple)):
```
Comment (clbarnes) [inline src/python/module/z5py/dataset.py]: TypeError?

File: src/python/module/z5py/file.py [hunk 3]
```diff
@@ -105,13 +109,14 @@ def _check_n5_version(path):
...
-        assert key not in self.keys(), \
-            "z5py.File.create_group: Group is already existing"
+        if key in self:
+            raise RuntimeError("Group %s is already existing" % key)
         path = os.path.join(self.path, key)
         return Group.make_group(path, self.is_zarr, self.mode)
 
     def __getitem__(self, key):
-        assert key in self, "z5py.File.__getitem__: key does not exxist"
+        if key not in self:
```
Comment (clbarnes) [inline src/python/module/z5py/file.py]: KeyError?

File: src/python/module/z5py/group.py [hunk 4]
```diff
@@ -19,12 +19,14 @@ def open_group(cls, path, is_zarr, mode):
...
     def create_group(self, key):
-        assert key not in self.keys(), "Group is already existing"
+        if key in self.keys():
+            raise RuntimeError("Group %s is already existing" % key)
         path = os.path.join(self.path, key)
         return Group.make_group(path, self.is_zarr, self.mode)
 
     def __getitem__(self, key):
-        assert key in self, "z5py.File.__getitem__: key does not exxist"
+        if key not in self:
```
Comment (clbarnes) [inline src/python/module/z5py/group.py]: KeyError

**Comment by constantinpape** [comment]
Thanks for the comments, I chose not to use `KeyErrors`, because there is not a direct translation from a native C++ exception.
However, `pybind11`, which we use for python bindings, defines some custom exceptions that are translated to a `KeyError`.
I will consider changing it.

**Pull request merged**